### PR TITLE
Move package versioning to the release workflow

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
This pull request proposes keeping track of versions through GitHub Releases and using `package.json` numbers only for publishing. Additionally, we could generate brief release notes with [release-drafter](https://github.com/iterative/dvc/blob/master/.github/release-drafter.yml), like in the DVC repository.

I was wondering if this is just a hack or could help us to automate the release chores.

### CODE REVIEW

**Comment by 0x2b3bfa0** [comment]
If we manage releases through [GitHub Releases](https://docs.github.com/en/github/administering-a-repository/managing-releases-in-a-repository) and a bit of [ClickOps](https://twitter.com/mxcia/status/1244938574832775168), tags would be created automatically under the hood. Combining that with [release-drafter](https://github.com/release-drafter/release-drafter), the whole process would be nearly automatic, apart from approval and manual additions to release notes.

**Comment by 0x2b3bfa0** [comment]
> _We'd need to sort out Git tags first I suppose._

Do you mean sorting out the stale tags we have now?

**Comment by casperdcl** [comment]
Yes - otherwise `npm version ${{ github.ref }}` won't work because GitHub Releases won't work because we need tags!

**Comment by casperdcl** [comment]
- this repo currently doesn't use tags
  - we need to discuss why not & change that (CC @DavidGOrtega)
- there are different ways to create tags (manual `git tag`, CI `npm version`, GH Releases) and trigger release deployments

**Comment by 0x2b3bfa0** [comment]
> _there are different ways to create tags (git tag, npm version, GH Releases) and trigger release deployments. It also matters whether tags are created on the CI or locally._

🙏  Thanks for the clarification! I was thinking about managing tags automatically through GitHub Releases, so they get created each time a release is published and workflows get triggered on that event.

**Comment by DavidGOrtega** [comment]
> we need to discuss why not & change that (CC @DavidGOrtega)
 
Yep, we should discuss everything indeed, regarding always to a change like this.
At the very beginning I wanted to do gitflow since I use it a lot, having master for releases and develop as default, but was not accepted, indeed gitflow is not welcome in our company preferences.

Gitflow does not solve anything yet automatically, in other projects that I used it Ihad to publish to NPM the bump was still manually but we could implement it with the tags event pretty easy.

As @casperdcl says Im not also a big fan of automatic release notes and here is where the release manager is to me a brilliant figure.

 - Reviews the PRs leaving the rest of the team focus
 - Prepare the release

Its not a huge task so can be done once a week with every release and a daily time to review PRs

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - run: npm install
+      - run: npm version "$(basename ${{ github.ref }})"
```
Comment (DavidGOrtega) [inline .github/workflows/release.yml:12]: To allow this the release should be done using tags triggered by tags event

Comment (casperdcl) [reply .github/workflows/release.yml:12]: also maybe better to use

```suggestion
      - run: npm version "$(git describe --tags)"
```

to support `on.workflow_dispatch`?

Comment (0x2b3bfa0) [reply .github/workflows/release.yml:12]: Not neccessarily: [`release`](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#release) does also work and integrates a bit better with GitHub.

Comment (0x2b3bfa0) [reply .github/workflows/release.yml:12]: Using `on.workflow_dispatch` was just a matter of convenience for testing, but should probably be removed if we plan to automate this part of the release process.

**Comment by DavidGOrtega** [comment]
@casperdcl @0x2b3bfa0 to be honest Im a bit lost right now who is triggering this and when.

**Comment by 0x2b3bfa0** [comment]
> _In github you create a release when you create a tag_

As far as I know, you automatically create a tag each time you create a release. Doing it the other way around would require a workflow.

**Comment by casperdcl** [comment]
either:

1. manual tag -> CI deploys & also drafts GH release -> manual GH release notes approval
2. manual GH release notes -> GH creates tag -> CI deploys

I think (2) is risky since CI could fail after a release published, while (1) still allows for force-pushing tags to fix silly minor mistakes

**Comment by casperdcl** [comment]
Yeah it looked like you were merging things directly into master so I deleted `devel`. Forgot to update this PR's base branch before doing so :/

**Comment by casperdcl** [comment]
There you go :)

File: .github/workflows/release.yml (hunk 1 above)
Comment (casperdcl) [inline .github/workflows/release.yml:12]: tagging not to be automated (i.e. `npm version M.m.p && git push && git push --tags` to be run locally) for now

```suggestion
```

File: package-lock.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@dvcorg/cml",
-  "version": "0.3.8",
+  "version": "0.0.0-development",
```
Comment (casperdcl) [inline package-lock.json:3]:
```suggestion
  "version": "0.3.8",
```

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@dvcorg/cml",
-  "version": "0.3.8",
+  "version": "0.0.0-development",
```
Comment (casperdcl) [inline package.json:3]:
```suggestion
  "version": "0.3.8",
```

**Review by casperdcl — CHANGES_REQUESTED** [review]
turns out it's a one-line PR :)

**Comment by 0x2b3bfa0** [comment]
Actually a zero-line pull request 🤣

**Comment by DavidGOrtega** [comment]
@0x2b3bfa0 this wonk work. Jobs in the workflow runs not in order so you should wait in build.yaml for the pacakge job to be resolved

**Review by DavidGOrtega — COMMENTED** [review]
Please verify that the build package happens before the other

**Comment by 0x2b3bfa0** [comment]
https://github.com/iterative/cml/blob/3736d7f832f25d6a0a0dad70d26fd16846923c28/.github/workflows/release.yml#L37
Isn't the above code enough to achieve that?

**Comment by DavidGOrtega** [comment]
😅 I swear I looked for it in the workflow. I need to rest

**Comment by DavidGOrtega** [comment]
> Yes, because both #504 and #505 were hotfixes to the build system (?) and didn't qualify for a release (!)

hotfix/ convention is widely use to fast fix ASAP master in gitflow convention, we needed to fix that in order to make any other to pass the tests, thats why its a hotfix

**Pull request closed without merging**